Kudu.Core/Functions/FunctionManager.cs

```diff
@@ -262,64 +262,60 @@ public void DeleteFunction(string name)
 
         private FunctionEnvelope CreateFunctionConfig(string configContent, string functionName)
         {
-            var config = JObject.Parse(configContent);
+            var functionConfig = JObject.Parse(configContent);
+
             return new FunctionEnvelope
             {
                 Name = functionName,
                 ScriptRootPathHref = FilePathToVfsUri(GetFunctionPath(functionName), isDirectory: true),
-                ScriptHref = FilePathToVfsUri(GetFunctionScriptPath(functionName, config)),
+                ScriptHref = FilePathToVfsUri(GetFunctionScriptPath(functionName, functionConfig)),
                 ConfigHref = FilePathToVfsUri(GetFunctionConfigPath(functionName)),
                 SecretsFileHref = FilePathToVfsUri(GetFunctionSecretsFilePath(functionName)),
                 Href = GetFunctionHref(functionName),
-                Config = config,
+                Config = functionConfig,
                 TestData = GetFunctionTestData(functionName)
             };
         }
 
-        // Logic for this function is copied from here
-        // https://github.com/Azure/azure-webjobs-sdk-script/blob/e0a783e882dd8680bf23e3c8818fb9638071c21d/src/WebJobs.Script/Config/ScriptHost.cs#L113-L150
-        private string GetFunctionScriptPath(string functionName, JObject functionInfo)
+        private string GetFunctionScriptPath(string functionName, JObject functionConfig)
         {
             var functionPath = GetFunctionPath(functionName);
             var functionFiles = FileSystemHelpers.GetFiles(functionPath, "*.*", SearchOption.TopDirectoryOnly)
                 .Where(p => Path.GetFileName(p).ToLowerInvariant() != "function.json").ToArray();
 
-            if (functionFiles.Length == 0)
-            {
-                return functionPath;
-            }
-            else if (functionFiles.Length == 1)
+            return DeterminePrimaryScriptFile(functionConfig, functionFiles);
+        }
+
+        // Logic for this function is copied from here:
+        // https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Host/ScriptHost.cs
+        // These two implementations must stay in sync!
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
```

Comment:
This function was copied verbatim from the Functions repo. Unit tests for it are in that repo.